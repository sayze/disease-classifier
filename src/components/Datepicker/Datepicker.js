import React from 'react'
import { MuiPickersUtilsProvider, DatePicker } from '@material-ui/pickers'
import DateFnsUtils from '@date-io/date-fns'
import PropTypes from 'prop-types'

const Datepicker = props => (
  <MuiPickersUtilsProvider utils={DateFnsUtils}>
    <DatePicker
      autoOk
      variant="inline"
      format="dd/MM/yyyy"
      inputVariant="outlined"
      margin="normal"
      label="Date picker inline"
      {...props}
    />
  </MuiPickersUtilsProvider>
)

Datepicker.defaultProps = {
  props: {},
}

Datepicker.propTypes = {
  props: PropTypes.object,
}

export default Datepicker
