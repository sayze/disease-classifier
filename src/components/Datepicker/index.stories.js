import React from 'react'
import Datepicker from '.'

export default {
  title: 'Date Picker',
  component: Datepicker,
}

export const Default = () => <Datepicker />
