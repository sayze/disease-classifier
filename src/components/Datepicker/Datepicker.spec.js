import React from 'react'
import { mount } from 'enzyme'
import Datepicker from '.'

describe('Datepicker', () => {
  it('should render default date picker', () => {
    const component = mount(<Datepicker />)
    expect(component).toMatchSnapshot()
  })
})
