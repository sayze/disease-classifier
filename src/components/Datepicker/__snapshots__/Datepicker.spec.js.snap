// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Datepicker should render default date picker 1`] = `
<Datepicker
  props={Object {}}
>
  <MuiPickersUtilsProvider
    utils={[Function]}
  >
    <PickerWithState
      allowKeyboardControl={true}
      autoOk={true}
      format="yyyy/MM/dd"
      helperText=" "
      inputVariant="outlined"
      invalidDateMessage="Invalid Date Format"
      label="Date picker inline"
      margin="normal"
      maxDate={2100-01-01T00:00:00.000Z}
      maxDateMessage="Date should not be after maximal date"
      minDate={1900-01-01T00:00:00.000Z}
      minDateMessage="Date should not be before minimal date"
      openTo="date"
      props={Object {}}
      variant="inline"
      views={
        Array [
          "year",
          "date",
        ]
      }
    >
      <Wrapper
        DateInputProps={
          Object {
            "inputValue": "2020/05/10",
            "openPicker": [Function],
            "validationError": "",
          }
        }
        InputComponent={[Function]}
        format="yyyy/MM/dd"
        helperText=" "
        inputVariant="outlined"
        label="Date picker inline"
        margin="normal"
        onAccept={[Function]}
        onClear={[Function]}
        onDismiss={[Function]}
        onSetToday={[Function]}
        open={false}
        props={Object {}}
        variant="inline"
      >
        <InlineWrapper
          DateInputProps={
            Object {
              "inputValue": "2020/05/10",
              "openPicker": [Function],
              "validationError": "",
            }
          }
          InputComponent={[Function]}
          format="yyyy/MM/dd"
          helperText=" "
          inputVariant="outlined"
          label="Date picker inline"
          margin="normal"
          onAccept={[Function]}
          onClear={[Function]}
          onDismiss={[Function]}
          onSetToday={[Function]}
          open={false}
          props={Object {}}
        >
          <PureDateInput
            format="yyyy/MM/dd"
            helperText=" "
            inputRef={
              Object {
                "current": <input
                  aria-invalid="false"
                  class="MuiInputBase-input MuiOutlinedInput-input"
                  readonly=""
                  type="text"
                  value="2020/05/10"
                />,
              }
            }
            inputValue="2020/05/10"
            inputVariant="outlined"
            label="Date picker inline"
            margin="normal"
            openPicker={[Function]}
            props={Object {}}
            validationError=""
          >
            <WithStyles(ForwardRef(TextField))
              InputProps={
                Object {
                  "readOnly": true,
                }
              }
              error={false}
              format="yyyy/MM/dd"
              helperText=" "
              inputRef={
                Object {
                  "current": <input
                    aria-invalid="false"
                    class="MuiInputBase-input MuiOutlinedInput-input"
                    readonly=""
                    type="text"
                    value="2020/05/10"
                  />,
                }
              }
              label="Date picker inline"
              margin="normal"
              onClick={[Function]}
              onKeyDown={[Function]}
              props={Object {}}
              value="2020/05/10"
              variant="outlined"
            >
              <ForwardRef(TextField)
                InputProps={
                  Object {
                    "readOnly": true,
                  }
                }
                classes={
                  Object {
                    "root": "MuiTextField-root",
                  }
                }
                error={false}
                format="yyyy/MM/dd"
                helperText=" "
                inputRef={
                  Object {
                    "current": <input
                      aria-invalid="false"
                      class="MuiInputBase-input MuiOutlinedInput-input"
                      readonly=""
                      type="text"
                      value="2020/05/10"
                    />,
                  }
                }
                label="Date picker inline"
                margin="normal"
                onClick={[Function]}
                onKeyDown={[Function]}
                props={Object {}}
                value="2020/05/10"
                variant="outlined"
              >
                <WithStyles(ForwardRef(FormControl))
                  className="MuiTextField-root"
                  color="primary"
                  disabled={false}
                  error={false}
                  format="yyyy/MM/dd"
                  fullWidth={false}
                  margin="normal"
                  onClick={[Function]}
                  onKeyDown={[Function]}
                  props={Object {}}
                  required={false}
                  variant="outlined"
                >
                  <ForwardRef(FormControl)
                    className="MuiTextField-root"
                    classes={
                      Object {
                        "fullWidth": "MuiFormControl-fullWidth",
                        "marginDense": "MuiFormControl-marginDense",
                        "marginNormal": "MuiFormControl-marginNormal",
                        "root": "MuiFormControl-root",
                      }
                    }
                    color="primary"
                    disabled={false}
                    error={false}
                    format="yyyy/MM/dd"
                    fullWidth={false}
                    margin="normal"
                    onClick={[Function]}
                    onKeyDown={[Function]}
                    props={Object {}}
                    required={false}
                    variant="outlined"
                  >
                    <div
                      className="MuiFormControl-root MuiTextField-root MuiFormControl-marginNormal"
                      format="yyyy/MM/dd"
                      onClick={[Function]}
                      onKeyDown={[Function]}
                      props={Object {}}
                    >
                      <WithStyles(ForwardRef(InputLabel))>
                        <ForwardRef(InputLabel)
                          classes={
                            Object {
                              "animated": "MuiInputLabel-animated",
                              "asterisk": "MuiInputLabel-asterisk",
                              "disabled": "Mui-disabled",
                              "error": "Mui-error",
                              "filled": "MuiInputLabel-filled",
                              "focused": "Mui-focused",
                              "formControl": "MuiInputLabel-formControl",
                              "marginDense": "MuiInputLabel-marginDense",
                              "outlined": "MuiInputLabel-outlined",
                              "required": "Mui-required",
                              "root": "MuiInputLabel-root",
                              "shrink": "MuiInputLabel-shrink",
                            }
                          }
                        >
                          <WithStyles(ForwardRef(FormLabel))
                            className="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined"
                            classes={
                              Object {
                                "asterisk": "MuiInputLabel-asterisk",
                                "disabled": "Mui-disabled",
                                "error": "Mui-error",
                                "focused": "Mui-focused",
                                "required": "Mui-required",
                              }
                            }
                            data-shrink={true}
                          >
                            <ForwardRef(FormLabel)
                              className="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined"
                              classes={
                                Object {
                                  "asterisk": "MuiFormLabel-asterisk MuiInputLabel-asterisk",
                                  "colorSecondary": "MuiFormLabel-colorSecondary",
                                  "disabled": "Mui-disabled Mui-disabled",
                                  "error": "Mui-error Mui-error",
                                  "filled": "MuiFormLabel-filled",
                                  "focused": "Mui-focused Mui-focused",
                                  "required": "Mui-required Mui-required",
                                  "root": "MuiFormLabel-root",
                                }
                              }
                              data-shrink={true}
                            >
                              <label
                                className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined MuiFormLabel-filled"
                                data-shrink={true}
                              >
                                Date picker inline
                              </label>
                            </ForwardRef(FormLabel)>
                          </WithStyles(ForwardRef(FormLabel))>
                        </ForwardRef(InputLabel)>
                      </WithStyles(ForwardRef(InputLabel))>
                      <WithStyles(ForwardRef(OutlinedInput))
                        autoFocus={false}
                        fullWidth={false}
                        inputRef={
                          Object {
                            "current": <input
                              aria-invalid="false"
                              class="MuiInputBase-input MuiOutlinedInput-input"
                              readonly=""
                              type="text"
                              value="2020/05/10"
                            />,
                          }
                        }
                        label={
                          <React.Fragment>
                            Date picker inline
                          </React.Fragment>
                        }
                        multiline={false}
                        readOnly={true}
                        value="2020/05/10"
                      >
                        <ForwardRef(OutlinedInput)
                          autoFocus={false}
                          classes={
                            Object {
                              "adornedEnd": "MuiOutlinedInput-adornedEnd",
                              "adornedStart": "MuiOutlinedInput-adornedStart",
                              "colorSecondary": "MuiOutlinedInput-colorSecondary",
                              "disabled": "Mui-disabled",
                              "error": "Mui-error",
                              "focused": "Mui-focused",
                              "input": "MuiOutlinedInput-input",
                              "inputAdornedEnd": "MuiOutlinedInput-inputAdornedEnd",
                              "inputAdornedStart": "MuiOutlinedInput-inputAdornedStart",
                              "inputMarginDense": "MuiOutlinedInput-inputMarginDense",
                              "inputMultiline": "MuiOutlinedInput-inputMultiline",
                              "marginDense": "MuiOutlinedInput-marginDense",
                              "multiline": "MuiOutlinedInput-multiline",
                              "notchedOutline": "MuiOutlinedInput-notchedOutline",
                              "root": "MuiOutlinedInput-root",
                            }
                          }
                          fullWidth={false}
                          inputRef={
                            Object {
                              "current": <input
                                aria-invalid="false"
                                class="MuiInputBase-input MuiOutlinedInput-input"
                                readonly=""
                                type="text"
                                value="2020/05/10"
                              />,
                            }
                          }
                          label={
                            <React.Fragment>
                              Date picker inline
                            </React.Fragment>
                          }
                          multiline={false}
                          readOnly={true}
                          value="2020/05/10"
                        >
                          <WithStyles(ForwardRef(InputBase))
                            autoFocus={false}
                            classes={
                              Object {
                                "adornedEnd": "MuiOutlinedInput-adornedEnd",
                                "adornedStart": "MuiOutlinedInput-adornedStart",
                                "colorSecondary": "MuiOutlinedInput-colorSecondary",
                                "disabled": "Mui-disabled",
                                "error": "Mui-error",
                                "focused": "Mui-focused",
                                "input": "MuiOutlinedInput-input",
                                "inputAdornedEnd": "MuiOutlinedInput-inputAdornedEnd",
                                "inputAdornedStart": "MuiOutlinedInput-inputAdornedStart",
                                "inputMarginDense": "MuiOutlinedInput-inputMarginDense",
                                "inputMultiline": "MuiOutlinedInput-inputMultiline",
                                "marginDense": "MuiOutlinedInput-marginDense",
                                "multiline": "MuiOutlinedInput-multiline",
                                "notchedOutline": null,
                                "root": "MuiOutlinedInput-root",
                              }
                            }
                            fullWidth={false}
                            inputComponent="input"
                            inputRef={
                              Object {
                                "current": <input
                                  aria-invalid="false"
                                  class="MuiInputBase-input MuiOutlinedInput-input"
                                  readonly=""
                                  type="text"
                                  value="2020/05/10"
                                />,
                              }
                            }
                            multiline={false}
                            readOnly={true}
                            renderSuffix={[Function]}
                            type="text"
                            value="2020/05/10"
                          >
                            <ForwardRef(InputBase)
                              autoFocus={false}
                              classes={
                                Object {
                                  "adornedEnd": "MuiInputBase-adornedEnd MuiOutlinedInput-adornedEnd",
                                  "adornedStart": "MuiInputBase-adornedStart MuiOutlinedInput-adornedStart",
                                  "colorSecondary": "MuiInputBase-colorSecondary MuiOutlinedInput-colorSecondary",
                                  "disabled": "Mui-disabled Mui-disabled",
                                  "error": "Mui-error Mui-error",
                                  "focused": "Mui-focused Mui-focused",
                                  "formControl": "MuiInputBase-formControl",
                                  "fullWidth": "MuiInputBase-fullWidth",
                                  "input": "MuiInputBase-input MuiOutlinedInput-input",
                                  "inputAdornedEnd": "MuiInputBase-inputAdornedEnd MuiOutlinedInput-inputAdornedEnd",
                                  "inputAdornedStart": "MuiInputBase-inputAdornedStart MuiOutlinedInput-inputAdornedStart",
                                  "inputHiddenLabel": "MuiInputBase-inputHiddenLabel",
                                  "inputMarginDense": "MuiInputBase-inputMarginDense MuiOutlinedInput-inputMarginDense",
                                  "inputMultiline": "MuiInputBase-inputMultiline MuiOutlinedInput-inputMultiline",
                                  "inputTypeSearch": "MuiInputBase-inputTypeSearch",
                                  "marginDense": "MuiInputBase-marginDense MuiOutlinedInput-marginDense",
                                  "multiline": "MuiInputBase-multiline MuiOutlinedInput-multiline",
                                  "root": "MuiInputBase-root MuiOutlinedInput-root",
                                }
                              }
                              fullWidth={false}
                              inputComponent="input"
                              inputRef={
                                Object {
                                  "current": <input
                                    aria-invalid="false"
                                    class="MuiInputBase-input MuiOutlinedInput-input"
                                    readonly=""
                                    type="text"
                                    value="2020/05/10"
                                  />,
                                }
                              }
                              multiline={false}
                              readOnly={true}
                              renderSuffix={[Function]}
                              type="text"
                              value="2020/05/10"
                            >
                              <div
                                className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl"
                                onClick={[Function]}
                              >
                                <input
                                  aria-invalid={false}
                                  autoFocus={false}
                                  className="MuiInputBase-input MuiOutlinedInput-input"
                                  disabled={false}
                                  onAnimationStart={[Function]}
                                  onBlur={[Function]}
                                  onChange={[Function]}
                                  onFocus={[Function]}
                                  readOnly={true}
                                  required={false}
                                  type="text"
                                  value="2020/05/10"
                                />
                                <WithStyles(ForwardRef(NotchedOutline))
                                  className="MuiOutlinedInput-notchedOutline"
                                  label={
                                    <React.Fragment>
                                      Date picker inline
                                    </React.Fragment>
                                  }
                                  labelWidth={0}
                                  notched={true}
                                >
                                  <ForwardRef(NotchedOutline)
                                    className="MuiOutlinedInput-notchedOutline"
                                    classes={
                                      Object {
                                        "legend": "PrivateNotchedOutline-legend-60",
                                        "legendLabelled": "PrivateNotchedOutline-legendLabelled-61",
                                        "legendNotched": "PrivateNotchedOutline-legendNotched-62",
                                        "root": "PrivateNotchedOutline-root-59",
                                      }
                                    }
                                    label={
                                      <React.Fragment>
                                        Date picker inline
                                      </React.Fragment>
                                    }
                                    labelWidth={0}
                                    notched={true}
                                  >
                                    <fieldset
                                      aria-hidden={true}
                                      className="PrivateNotchedOutline-root-59 MuiOutlinedInput-notchedOutline"
                                    >
                                      <legend
                                        className="PrivateNotchedOutline-legendLabelled-61 PrivateNotchedOutline-legendNotched-62"
                                      >
                                        <span>
                                          Date picker inline
                                        </span>
                                      </legend>
                                    </fieldset>
                                  </ForwardRef(NotchedOutline)>
                                </WithStyles(ForwardRef(NotchedOutline))>
                              </div>
                            </ForwardRef(InputBase)>
                          </WithStyles(ForwardRef(InputBase))>
                        </ForwardRef(OutlinedInput)>
                      </WithStyles(ForwardRef(OutlinedInput))>
                      <WithStyles(ForwardRef(FormHelperText))>
                        <ForwardRef(FormHelperText)
                          classes={
                            Object {
                              "contained": "MuiFormHelperText-contained",
                              "disabled": "Mui-disabled",
                              "error": "Mui-error",
                              "filled": "MuiFormHelperText-filled",
                              "focused": "Mui-focused",
                              "marginDense": "MuiFormHelperText-marginDense",
                              "required": "Mui-required",
                              "root": "MuiFormHelperText-root",
                            }
                          }
                        >
                          <p
                            className="MuiFormHelperText-root MuiFormHelperText-contained MuiFormHelperText-filled"
                          >
                            <span
                              dangerouslySetInnerHTML={
                                Object {
                                  "__html": "&#8203;",
                                }
                              }
                            />
                          </p>
                        </ForwardRef(FormHelperText)>
                      </WithStyles(ForwardRef(FormHelperText))>
                    </div>
                  </ForwardRef(FormControl)>
                </WithStyles(ForwardRef(FormControl))>
              </ForwardRef(TextField)>
            </WithStyles(ForwardRef(TextField))>
          </PureDateInput>
          <WithStyles(ForwardRef(Popover))
            anchorOrigin={
              Object {
                "horizontal": "center",
                "vertical": "bottom",
              }
            }
            onClose={[Function]}
            open={false}
            transformOrigin={
              Object {
                "horizontal": "center",
                "vertical": "top",
              }
            }
          >
            <ForwardRef(Popover)
              anchorOrigin={
                Object {
                  "horizontal": "center",
                  "vertical": "bottom",
                }
              }
              classes={
                Object {
                  "paper": "MuiPopover-paper",
                  "root": "MuiPopover-root",
                }
              }
              onClose={[Function]}
              open={false}
              transformOrigin={
                Object {
                  "horizontal": "center",
                  "vertical": "top",
                }
              }
            >
              <ForwardRef(Modal)
                BackdropProps={
                  Object {
                    "invisible": true,
                  }
                }
                className="MuiPopover-root"
                onClose={[Function]}
                open={false}
              />
            </ForwardRef(Popover)>
          </WithStyles(ForwardRef(Popover))>
        </InlineWrapper>
      </Wrapper>
    </PickerWithState>
  </MuiPickersUtilsProvider>
</Datepicker>
`;
